core/unix/src/TUnixSystem.cxx

```diff
@@ -40,10 +40,15 @@
 #include <algorithm>
 #include <atomic>
 
+#ifdef __linux__
+#include <syscall.h>
+#endif
+
 //#define G__OLDEXPAND
 
 #include <unistd.h>
 #include <stdlib.h>
+#include <poll.h>
```

Comment:
Nope, `fSignals` doesn't need to be changed because it isn't invoked from a signal handler.

Unix signal handlers have an extremely limited set of operations they are allowed to do: this is why we must be very careful to not call most ROOT functions.